refactor(oxlint): add explicit return type, and remove jsdoc type

camc314 opened the pull request:
since we use TS files, we should just use type annotations rather than JSDoc annotations

overlookmotel reviewed (approved):
I guess I missed that one...

camc314 committed:
refactor(oxlint): add explicit return type, and remove jsdoc type (#15909)

since we use TS files, we should just use type annotations rather than JSDoc annotations
